miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultLibraryService.java

```diff
@@ -363,15 +361,13 @@ public void addQc(Library library, LibraryQC qc) throws IOException {
     Library managed = libraryDao.get(library.getId());
     authorizationManager.throwIfNotWritable(managed);
     qc.setQcCreator(authorizationManager.getCurrentUsername());
-    try {
-      qc.setLibrary(managed);
-      managed.addQc(qc);
-    } catch (MalformedLibraryException e) {
-      throw new IOException("Malformed Library " + managed.getName() + " in the database...", e);
-    } catch (MalformedLibraryQcException e) {
-      throw new IOException("Malformed Library QC");
+
+    // update concentration if QC is of relevant type
+    if ("QuBit".equals(qc.getQcType().getName())) {
```

Comment:
Would be better if there were a field in the QcType table to indicate purpose=concentration rather than hard-coding something that may or may not exist in the DB. Maybe this is okay temporarily